NitroxClient/GameLogic/Spawning/WorldEntities/PlaceholderGroupWorldEntitySpawner.cs

```diff
@@ -1,16 +1,26 @@
 using System.Collections;
+using System.Linq;
 using NitroxClient.GameLogic.Spawning.Metadata;
 using NitroxClient.MonoBehaviours;
 using NitroxModel.DataStructures.GameLogic;
 using NitroxModel.DataStructures.GameLogic.Entities;
 using NitroxModel.DataStructures.Util;
 using UnityEngine;
-using UWE;
 
 namespace NitroxClient.GameLogic.Spawning.WorldEntities;
 
+/// <summary>
+/// Group placeholders are unique in that the game cleans up their children each time a player moves out of its batch. In single player, the placeholders
+/// would be respawned by iterating over the placeholders - this is patched out in nitrox because we need to tag them. Due to this behavior, we need to 
+/// respawn the placeholders when the batch is reloaded.  The top level group entity will still exist in the batch; thus, we will try to load the object
+/// by ID first and if we can't find it then spawn the entity in.  
+/// </summary>
 public class PlaceholderGroupWorldEntitySpawner : IWorldEntitySpawner
 {
+    // A simple way to detect if the respawn is triggered from a batch deserialization is to tag the group with a mono that is not registered in protobuf.
+    // If the group is untagged on a subsequent call then we know to respawn children again. 
+    private class BatchRespawnDetection : MonoBehaviour { };
```

Comment:
Would call it `NitroxBatchRespawnDetection` so its origin is clear in the SceneDebugger.